plugins/utils/autosource.nix

```
@@ -0,0 +1,24 @@
+{
+  helpers,
+  ...
+}:
+helpers.vim-plugin.mkVimPlugin {
+  name = "autosource";
+  originalName = "vim-autosource";
+  package = "vim-autosource";
+  globalPrefix = "autosource_";
+
+  maintainers = [ helpers.maintainers.refaelsh ];
+
+  settingsOptions = {
+    autosource_prompt_for_new_file = helpers.defaultNullOpts.mkBool true ''
```

Settings options shouldn't include the globalsPrefix.

```suggestion
    prompt_for_new_file = defaultNullOpts.mkEnum [ 0 1 ] 1 ''
```

Also, this is a 0 or 1 type, not a bool. See #1741 and #1834